curl.ml

```diff
@@ -324,6 +324,8 @@ type curlPostRedir =
 | REDIR_POST_302
 | REDIR_POST_303
 
+type curlWritefuncOpt = string -> [`Full | `Partial of int | `Pause]
```

Comment:
lets use normal variants, not polymorphic.
I propose `Pause | Abort | Continue` (couldn't find better name than Continue (it may confuse people to think it has a meaning of "resume", which it does not), ideas welcome)